packages/design-system/src/components/popup/index.js

```diff
@@ -20,98 +20,124 @@
 import styled from 'styled-components';
 import {
   useLayoutEffect,
+  useEffect,
   useCallback,
   useState,
   useRef,
   useResizeEffect,
   createPortal,
-  useEffect,
 } from '@googleforcreators/react';
 import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
-import { themeHelpers } from '../../theme';
 import { noop } from '../../utils';
 import { getTransforms, getOffset } from './utils';
 import { PLACEMENT } from './constants';
-
-// TODO scrollbar update, commented out until design updates are done
-const DEFAULT_POPUP_Z_INDEX = 11;
-
+const DEFAULT_TOPOFFSET = 0;
 const Container = styled.div.attrs(
   ({
     $offset: { x, y, width, height },
     fillWidth,
     fillHeight,
     placement,
-    zIndex,
+    isRTL,
+    invisible,
   }) => ({
     style: {
-      transform: `translate(${x}px, ${y}px) ${getTransforms(placement)}`,
+      transform: `translate(${x}px, ${y}px) ${getTransforms(placement, isRTL)}`,
       ...(fillWidth ? { width: `${width}px` } : {}),
       ...(fillHeight ? { height: `${height}px` } : {}),
-      zIndex,
+      ...(invisible ? { visibility: `hidden` } : {}),
     },
   })
 )`
   /*! @noflip */
   left: 0px;
   top: 0px;
   position: fixed;
-
-  ${themeHelpers.scrollbarCSS};
+  z-index: 2;
+  ${({ noOverFlow }) => (noOverFlow ? `` : `overflow-y: auto;`)};
+  max-height: ${({ topOffset }) => `calc(100vh - ${topOffset}px)`};
 `;
-
 function Popup({
+  isRTL = false,
   anchor,
   dock,
   children,
   renderContents,
   placement = PLACEMENT.BOTTOM,
-  zIndex = DEFAULT_POPUP_Z_INDEX,
   spacing,
   isOpen,
+  invisible = false,
   fillWidth = false,
   fillHeight = false,
   onPositionUpdate = noop,
   refCallback = noop,
+  topOffset = DEFAULT_TOPOFFSET,
+  noOverFlow = false,
+  shouldHaveDifferentOffset = false,
 }) {
   const [popupState, setPopupState] = useState(null);
   const isMounted = useRef(false);
   const popup = useRef(null);
 
   useEffect(() => {
     isMounted.current = true;
-
     return () => {
       isMounted.current = false;
     };
   }, []);
 
   const positionPopup = useCallback(
-    async (evt) => {
-      if (!isMounted.current) {
+    (evt) => {
+      if (!isMounted.current || !anchor?.current) {
         return;
       }
-
       // If scrolling within the popup, ignore.
       if (evt?.target?.nodeType && popup.current?.contains(evt.target)) {
         return;
       }
-
-      await setPopupState({
-        offset:
-          anchor?.current && getOffset(placement, spacing, anchor, dock, popup),
+      setPopupState({
+        offset: anchor?.current
```

Comment:
nit: We don't need the optional chaining here since we already check for it above. `anchor.current` is fine.